docs/root/configuration/http/http_filters/router_filter.rst

```diff
@@ -374,6 +374,19 @@ compared to network latency between client and Envoy. This header is set on resp
 
 .. _config_http_filters_router_x-envoy-overloaded_set:
 
+x-envoy-upstream-stream-duration-ms
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This value is used to configure the maximum stream lifetime for the stream which
+has this header. If the stream exceeds this lifetime, it will be reset and a 408 response
+will be sent to downstream. If the value of the header is 0, then the lifetime will be
+infinite and no limit will be enforce. It is similar to
+:ref:`max_stream_duration <envoy_v3_api_field_config.core.v3.HttpProtocolOptions.max_stream_duration>`,
+but that configuration applies to all streams on the route. If set, this header will
+override the cluster configuration.
```

Comment:
looks like you're mixing routes and clusters here?

I think there's max_stream_duration set for HCM and route (downstream) and max_stream_duration set for the cluster (upstream).  AFIK They operate pretty much independently.  I think this header only overrides the upstream one which I assume is the intent given the name, but given the complexity here I think we can be a bit more clear in comments.

This value is used to configure the maximum stream lifetime for the stream which has this header. ->
This value is used to configure the maximum _upstream_ stream lifetime for the stream which has this header.

but that configuration applies to all streams on the route. ->
but that configuration applies to all streams to this cluster.

Reply:
fixed